File: HarmonyOSExample/entry/src/main/java/com/minkiapps/hos/migration/util/LoggerFacade.java
package com.minkiapps.hos.migration.util;

import com.minkiapps.shared.util.ILogger;
import ohos.hiviewdfx.HiLog;
import ohos.hiviewdfx.HiLogLabel;
import org.jetbrains.annotations.NotNull;

public class LoggerFacade implements ILogger {

    private static final String TAG_LOG = "MigrationExample";

    private static final int DOMAIN_ID = 0x0001;

    private static final HiLogLabel LABEL_LOG = new HiLogLabel(HiLog.LOG_APP, DOMAIN_ID, LoggerFacade.TAG_LOG);

    private static final String LOG_FORMAT = "%{public}s: %{public}s";

    @Override
    public void debug(@NotNull String tag, @NotNull String message) {
        HiLog.debug(LABEL_LOG, LOG_FORMAT, tag, message);
    }

    @Override
    public void error(@NotNull String tag, @NotNull String message) {
        HiLog.error(LABEL_LOG, LOG_FORMAT, tag, message);
    }
}

File: AndroidExample/app/src/main/java/com/minkiapps/android/example/util/LoggerFacade.kt
package com.minkiapps.android.example.util

import android.util.Log
import com.minkiapps.shared.util.ILogger

class LoggerFacade : ILogger {

    override fun debug(tag: String, message: String) {
        Log.d(tag, message)
    }

    override fun error(tag: String, message: String) {
        Log.d(tag, message)
    }
}